### PR TITLE
feat!: rewire ucan endpoint back to /upload

### PR DESCRIPTION
**Pull request by Gozala**

### CODE REVIEW

**Review by alanshaw — CHANGES_REQUESTED** [review]
Just noticed:

https://github.com/nftstorage/nft.storage/blob/b29d3d5a431f4775fabd147bc66353956511e8f9/packages/api/src/errors.js#L238-L256

Can you fix `this.code` in both and add missing `ErrorAgentDIDRequired.CODE`.

File: packages/client/src/lib.js [hunk 1]
```diff
@@ -112,15 +113,26 @@ class NFTStorage {
...
 
   /**
    * @hidden
-   * @param {string} token
+   * @param {object} options
+   * @param {string} options.token
+   * @param {string} [options.did]
    */
-  static auth(token) {
+  static auth({ token, did }) {
```
Comment (alanshaw) [inline packages/client/src/lib.js:129]: Signature changed but none of the callers

Comment (Gozala) [reply packages/client/src/lib.js:129]: That is because I use spread to capture both e.g see line 170

Comment (alanshaw) [reply packages/client/src/lib.js:129]: Oh yes my apologies!

**Review by alanshaw — CHANGES_REQUESTED** [review]
`this.code` in `ErrorDIDNotFound` is still not correct!

File: packages/api/src/errors.js [hunk 2]
```diff
@@ -251,12 +251,7 @@ export class ErrorAgentDIDRequired extends HTTPError {
...
     super(msg, status)
     this.name = 'ErrorAgentDIDRequired'
-    this.code = ErrorUnauthenticated.CODE
-  }
-}
-
-export class ErrorInvalidRoute extends HTTPError {
-  constructor(msg = 'Invalid route for the request', status = 400) {
-    super(msg, status)
+    this.code = ErrorAgentDIDRequired.CODE
```
Comment (Gozala) [inline packages/api/src/errors.js:254]:
```suggestion
    this.name = 'AgentDIDRequired'
    this.code = ErrorAgentDIDRequired.CODE
```

File: packages/api/src/errors.js [hunk 3]
```diff
@@ -242,7 +242,7 @@ export class ErrorDIDNotFound extends HTTPError {
     this.code = ErrorMaintenance.CODE
```
Comment (Gozala) [inline packages/api/src/errors.js:242]:
```suggestion
    this.code = ErrorDIDNotFound.CODE
```

**Review by alanshaw — APPROVED** [review]
LGTM

**Pull request merged**